CascadeEnergy/consul-client
# SEN-1555 -- Support for targeting service versions via consul tags.

Will be a breaking change `3.0.0`

**To Test**

You need to deploy a versioned "users-service" to your dev account, here's a step by step on how to do that: https://github.com/CascadeEnergy/pym/pull/64

Now upload this repo to a box in your account which has access to consul.

`$ cd /var/cascade/users-service && mkdir tmp`
`$ vim tmp/test.js`

Copy/Pasta this: 

**REPLACE** `x.x.x.x` with the Private IP of your consul box.

```javascript
'use strict';

var consulClient = require('../');

var consulHost = 'x.x.x.x:8500'; // e.g 172.x.x.x:8500
var consulRequest = consulClient(consulHost);

var config = {
  serviceName: 'users',
  version: '1.0.0',
  endpoint: 'users'
};

// Discovers an instance of the users service
// and POSTs body to it's /users/login route.
consulRequest(config)
  .then(function(res) {
    console.log(JSON.stringify(res.body, null, 2));
  })
  .catch(console.log);
```

`$ node tmp/test.js`

should get something like this if you have a "users" dynamodb table in your account:

```json
{
  "Count": 1,
  "Items": [
    {
      "userId": "will"
    }
  ],
  "ScannedCount": 1
}
```
